src/libs/dutil/butil.cpp

```diff
@@ -214,15 +212,12 @@ HRESULT DAPI BundleEnumRelatedBundle(
 
         if (fUpgradeCodeFound)
         {
-            hr = RegReadString(hkBundle, BUNDLE_REGISTRATION_REGISTRY_BUNDLE_PROVIDER_KEY, &szProviderKey );
-            ExitOnFailure(hr, "Failed to read the bundle provider key.");
-
-            hr = ::StringCchLengthW(szProviderKey, STRSAFE_MAX_CCH, reinterpret_cast<UINT_PTR*>(&cchProviderKey));
-            ExitOnFailure(hr, "Failed to calculate length of string");
-
             if (lpBundleIdBuf)
             {
-                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, szProviderKey, cchProviderKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
+                hr = ::StringCchLengthW(sczUninstallSubKey, STRSAFE_MAX_CCH, reinterpret_cast<UINT_PTR*>(&cchUninstallSubKey));
+                ExitOnFailure(hr, "Failed to calculate length of string");
+
+                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, sczUninstallSubKey, cchUninstallSubKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
```

Comment:
Brain fart, and it's what I had previously done incorrectly.

Will update accordingly.

On Saturday, July 16, 2016, Bob Arnson <notifications@github.com> wrote:

> In src/libs/dutil/butil.cpp
> <https://github.com/wixtoolset/wix3/pull/369#discussion_r71072177>:
>
> >              if (lpBundleIdBuf)
> >              {
> > -                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, szProviderKey, cchProviderKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
> > +                hr = ::StringCchLengthW(sczUninstallSubKey, STRSAFE_MAX_CCH, reinterpret_cast<UINT_PTR*>(&cchUninstallSubKey));
> > +                ExitOnFailure(hr, "Failed to calculate length of string");
> > +
> > +                hr = ::StringCchCatNExW(lpBundleIdBuf, MAX_GUID_CHARS + 1, sczUninstallSubKey, cchUninstallSubKey, NULL, NULL, STRSAFE_FILL_BEHIND_NULL);
>
> StringCchCatNExW is going to fail after the first GUID if the limit is
> MAX_GUID_CHARS. Any reason not to use StringCchCopy*?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wixtoolset/wix3/pull/369/files/5633d66fdcf24762ae10d1951435f7cf2a7df1d6#r71072177>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACmlRUIIO3-S1z8raWOHnMoqM2kgHo6Iks5qWU3bgaJpZM4JI81z>
> .
>